flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSplitPlanner.java

```diff
@@ -137,6 +121,60 @@ static CloseableIterable<CombinedScanTask> planTasks(
     }
   }
 
+  private static Long resolveToSnapshotId(Table table, ScanContext context) {
+    if (context.endTag() != null) {
+      Preconditions.checkArgument(
```

Comment:
As @hililiwei mentioned, could we check these in the `ScanContext.validate()`?

Reply:
`ScanContext.validate()` can only verify the validity of the variables themselves, but cannot determine whether the `Tag` is correctly bound to an available snapshot id.